Make size of input[type="color"] aware of vertical writing mode

nt1m opened the pull request:
#### 8786fe417c93f5eb7009fe3a346002a09b922a8b
<pre>
Make size of input[type=&quot;color&quot;] aware of vertical writing mode
<a href="https://bugs.webkit.org/show_bug.cgi?id=248253">https://bugs.webkit.org/show_bug.cgi?id=248253</a>
rdar://102616389

Reviewed by Cameron McCormack.

Use logical property counterparts on desktop platforms. Also, enable testing of vertical writing mode controls.

* LayoutTests/TestExpectations:
* LayoutTests/platform/ios/TestExpectations:
* Source/WebCore/css/horizontalFormControls.css:
(@namespace &quot;<a href="http://www.w3.org/1999/xhtml&quot">http://www.w3.org/1999/xhtml&quot</a>;;):
* Source/WebCore/css/html.css:
(input:not([type=&quot;color&quot;]), textarea, select, button, meter, progress):
(#endif):
* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::RenderTheme::colorInputStyleSheet const):
* Tools/WebKitTestRunner/TestOptions.cpp:
(WTR::TestOptions::defaults):

Canonical link: <a href="https://commits.webkit.org/256967@main">https://commits.webkit.org/256967@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/50ad486211444c7214da28d3df67165fa351e3a0

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/97382 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/6644 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/30546 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/106900 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/167166 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/101348 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/6957 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/35388 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/89789 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/103577 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/103045 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/5222 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/84013 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/32223 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/87084 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/88895 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/75148 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/659 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/20361 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/641 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/21826 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4803 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/5447 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/44308 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [  ~~🛠 🧪 unsafe-merge~~](https://ews-build.webkit.org/#/builders/75/builds/2362 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/1887 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/41175 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

heycam commented on Source/WebCore/rendering/RenderThemeIOS.mm:
```diff
@@ -2278,7 +2278,7 @@ static void paintAttachmentBorder(GraphicsContext& context, Path& borderPath)
         return RenderTheme::colorInputStyleSheet(settings);
 
-    return "input[type=\"color\"] { appearance: auto; width: 28px; height: 28px; box-sizing: border-box; outline: none; border: initial; border-radius: 50%; } "_s;
+    return "input[type=\"color\"] { appearance: auto; inline-size: 28px; block-size: 28px; box-sizing: border-box; outline: none; border: initial; border-radius: 50%; } "_s;
```
I guess this isn't necessary (and we're not changing the `::-webkit-color-swatch` `padding` declaration to be logical for iOS).

nt1m replied:
Ah you're right! I didn't look very closely.

nt1m committed:
Make size of input[type="color"] aware of vertical writing mode
https://bugs.webkit.org/show_bug.cgi?id=248253
rdar://102616389

Reviewed by Cameron McCormack.

Use logical property counterparts on desktop platforms. Also, enable testing of vertical writing mode controls.

* LayoutTests/TestExpectations:
* LayoutTests/platform/ios/TestExpectations:
* Source/WebCore/css/horizontalFormControls.css:
(@namespace "http://www.w3.org/1999/xhtml";):
* Source/WebCore/css/html.css:
(input:not([type="color"]), textarea, select, button, meter, progress):
(#endif):
* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::RenderTheme::colorInputStyleSheet const):
* Tools/WebKitTestRunner/TestOptions.cpp:
(WTR::TestOptions::defaults):

Canonical link: https://commits.webkit.org/256967@main
